lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -0,0 +1,99 @@
+from http import HTTPStatus
+
+import pytest
+import requests
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources.datasets_inventory import DatasetsInventory
+from pbench.server.filetree import FileTree
+from pathlib import Path
+from pbench.server.database.models.datasets import Dataset, DatasetNotFound
+
+
+class TestDatasetsAccess:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config, more_datasets, provide_metadata):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            more_datasets: Dataset construction fixture
+            provide_metadata: Dataset metadata fixture
+        """
+
+        def query_api(
+            dataset: str, username: str, expected_status: HTTPStatus, path: str
+        ) -> requests.Response:
+            headers = None
+            try:
+                dataset = Dataset.query(name=dataset).resource_id
+            except DatasetNotFound:
+                dataset = dataset  # Allow passing deliberately bad value
+            if username:
+                token = self.token(client, server_config, username)
+                headers = {"authorization": f"bearer {token}"}
+            response = client.get(
+                f"{server_config.rest_uri}/datasets/inventory/{dataset}/{path}",
+                headers=headers,
+            )
+            assert response.status_code == expected_status
+
+            # We need to log out to avoid "duplicate auth token" errors on the
+            # "put" test which does a PUT followed by two GETs.
+            if username:
+                client.post(
+                    f"{server_config.rest_uri}/logout",
+                    headers={"authorization": f"bearer {token}"},
+                )
```

Comment:
Seems this was copied from `test_datasets_metadata.ps`, which had to deal with some special cases in both `PUT` and `GET` testing... which doesn't apply here.

In a previous comment, Webb commented about trying to reuse these query wrappers. I've thought about trying to refactor the unit test hierarchy again to pull out another level on top of (or rather underneath) what Nikhil factored out into the `Common` class. Never got around to doing it, though. (And honestly if I had that time and motivation I'd rather turn the API tests into true unit tests and true functional tests rather than preserve this mocked request half-state.)

Reply:
So...that's a "yes"?  😆